app/backend/src/tests/test_api.py

```diff
@@ -297,13 +331,23 @@ def test_friend_request_flow(db):
     with api_session(token1) as api:
         api.SendFriendRequest(api_pb2.SendFriendRequestReq(user_id=user2.id))
 
+    with session_scope() as session:
+        friend_request = (
+            session.query(FriendRelationship)
+            .filter(FriendRelationship.from_user_id == user1.id and FriendRelationship.to_user_id == user2.id)
+            .one_or_none()
+        )
+        session.expunge(friend_request)
```

Comment:
Please don't use expunge unless really necessary (even in tests): it causes a lot of errors that are hard to understand especially if one hasn't dealt with them before!

Instead just save the FR id as `friend_request_id = friend_request.id` and use that outside this scope